tests/SniffTest.php

```diff
@@ -45,6 +45,14 @@ public static function getComposerJson() {
                         'url' => 'https://download.civicrm.org/cv/cv.phar-2019-08-20-14fe9da8',
                         'path' => 'bin/cv',
                     ],
+                    'symfony-cli' => [
+                        'type' => 'archive',
+                        'version' => 'v5.4.13',
+                        'variables_provider' => '\\LastCall\\DownloadsPlugin\\Tests\\Fixtures\\SystemVariablesProvider',
+                        'url' => 'https://github.com/symfony-cli/symfony-cli/releases/download/{$version}/symfony-cli_{$os}_{$architecture}.{$extension}',
+                        'path' => 'extern/{$id}',
+                        'ignore' => ['LICENSE', 'README.md'],
+                    ],
```

Comment:
`symfony-cli` is chosen because it the only binary that I'm familiar with that has complex download url so I can put almost every variables in it. I don't know about civicrm so I don't know about other binaries that can replace it.

Reply:
That is an interesting example.

FWIW, `SniffTest` fails  on my laptop because it computes `symfony-cli_darwin_arm64.tar.gz` -- instead of the actual `symfony-cli_darwin_all.tar.gz`. But I guess that be fixed with another override in `SystemVariablesProvider`?